ledger/src/blockstore_meta.rs

```diff
@@ -526,6 +526,19 @@ impl OptimisticSlotMetaVersioned {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SizeDetails {
+    pub last_fec_set_index: u32,
+    pub num_coding_received: usize,
+    pub num_data_received: usize,
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum LastFecSetDetails {
```

Comment:
This is an enum to support future extensibility when we eventually add IP check/retransmit signature verification.

Reply:
Glad you already baked in an enum to allow it to grow. A few notes:
- One consideration will be backwards compat; namely, making sure old code that is unaware of new enum instances can handle those instances gracefully. Not a concern for this PR, but just mentioning it so we're thinking about it a little
- What will future enum instances of `LastFecSetDetails` look like. The current enum has just a `SizeDetails`; will future iterations contain a completely different struct OR `SizeDetails` + some new info OR something else?

Reply:
I envision future iterations to be inclusive, so `SizeDetails` + `Option<AdditionalInfo>`. 
We add additional info as an option in order to support backwards compatibility.